src/BinSkim.Rules/PERules/BA2006.BuildWithSecureTools.cs

```diff
@@ -258,33 +210,23 @@
 
                 if (foundIssue)
                 {
+                    if (!languageToBadModules.TryGetValue(omDetails.Language, out List<ObjectModuleDetails> badModules))
+                    {
+                        badModules = new List<ObjectModuleDetails>();
+                    }
+
                     badModules.Add(omDetails);
+                    languageToBadModules[omDetails.Language] = badModules;
```

Comment:
>languageToBadModules [](http://example.com/codeflow?start=20&length=20)

you are always setting bad modules back in the dictionary, instead of simply doing so when it couldn't be found. Move this code above into that TryGetValue case.